app/uk/gov/hmrc/nationalinsurancerecord/controllers/NationalInsuranceRecordController.scala

```diff
@@ -38,7 +38,7 @@ trait NationalInsuranceRecordController extends BaseController with HeaderValida
       content,
       selfLink,
       Some(Vector("taxYears" -> years.map(taxYear => halResourceSelfLink(Json.toJson(taxYear),
-        nationalInsuranceTaxYearHref(nino, taxYear.taxYear))).toVector))
+        nationalInsuranceTaxYearHref(nino, TaxYear(taxYear.taxYear)))).toVector))
```

Comment:
Is it worth updating TaxYearSummary to parse as a TaxYear rather than a String at the same time?